Simplify the user interface

frapac opened the pull request:
- add new constructors to SPModel types
- rename modules to be more explicit
- update dams.jl and damsvalley.jl
- improve coverage of unit-testings
- add a test based upon extended formulation

frapac committed:
[UPD] Change display param in solve_SDDP

+ bug hotfix in initialize_value_functions

frapac committed:
[UPD] Rename modules and functions

+ remove deprecated functions
+ update documentations of modules

frapac committed:
[DEV] Change constructors in objects.jl

frapac committed:
[UPD] Move extensiveFormulation to src

+ add this function to StochDynamicProgramming

frapac committed:
[DEV] Add unit-tests to improve coverage

- Test SDDP results with extensive formulation
- test multidimensional case

frapac committed:
[FIX] Fix bugs in damsvalley.jl

leclere commented on src/extensiveFormulation.jl:
```diff
@@ -7,47 +7,46 @@
-    
+
+
     #Define the conditional probabilities on each arc of the scenario tree
```
No, it computes the total probability of each node from the conditional probabilities.

frapac replied:
You should see that with Henri.

leclere commented on src/noises.jl:
```diff
@@ -83,14 +85,14 @@ Parameters:
 
 - n::Int
-    number of simulations computed
+    number of simulations to compute
```
scenarios

leclere commented on src/noises.jl:
```diff
@@ -83,14 +85,14 @@ Parameters:
-function simulate(law::Vector{NoiseLaw}, n::Int64)
+function generate_scenarios(law::Vector{NoiseLaw}, n::Int64)
     if n <= 0
         error("negative number of simulations")
```
scenarios

leclere commented on test/runtests.jl:
```diff
@@ -80,25 +87,142 @@ facts("SDDP algorithm") do
+        # Test a simulation upon given scenarios:
+        aleas = simulate_scenarios(model.noises,
+                              (model.stageNumber,
+                               params.forwardPassNumber,
```
Use n_simulations instead, else you are checking a monte carlo result with 10 simulation.

leclere commented on test/runtests.jl:
```diff
@@ -80,25 +87,142 @@ facts("SDDP algorithm") do
+        ef_cost = StochDynamicProgramming.extensive_formulation(model,params)
+        @fact typeof(ef_cost) --> Float64
+
+        @fact mean(sddp_costs) --> roughly(ef_cost)
```
Checking the lower bound might be more reliable.

frapac committed:
[UPD] Rescope PR27

- correct docstring
- use Monte-Carlo to estimate upper-bounds in runtests.jl
- fix comment in extensive_formulation.jl
